src/test/java/com/skynest/BucketContentTest.java

```diff
@@ -0,0 +1,123 @@
+package com.skynest;
+
+import com.skynest.models.CreateFolderRequest;
+import com.skynest.models.FileResponse;
+import com.skynest.models.FolderResponse;
+import com.skynest.models.StorageContentResponse;
+import io.restassured.response.Response;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.UUID;
+
+import static org.apache.http.HttpStatus.SC_FORBIDDEN;
+import static org.apache.http.HttpStatus.SC_OK;
+import static org.apache.http.HttpStatus.SC_NOT_FOUND;
+import static org.apache.http.HttpStatus.SC_BAD_REQUEST;
+import static org.testng.Assert.assertEquals;
+import static org.testng.AssertJUnit.assertNotNull;
+
+public class BucketContentTest extends BucketBaseTest {
+    @Test
+    void bucket_content_should_be_empty_when_folders_and_files_are_not_added() {
+        Response response = skyNestBackendClient.getBucketContent(createdBucketResponse.getBucketId());
+        response.then().statusCode(SC_OK);
+
+        StorageContentResponse getStorageContentResponse = response.as(StorageContentResponse.class);
+
+        assertEquals(getStorageContentResponse.getFolders().size(), 0);
+        assertEquals(getStorageContentResponse.getFiles().size(), 0);
+    }
+
+    @Test(dependsOnMethods = "bucket_content_should_be_empty_when_folders_and_files_are_not_added")
+    void logged_owner_should_be_able_to_create_folder_in_its_own_private_bucket() {
+        CreateFolderRequest createFolderRequest = CreateFolderRequest.generateValidFolderCreationRequest(createdBucketResponse.getBucketId());
+        Response response = skyNestBackendClient.createFolder(createFolderRequest);
+        response.then().statusCode(SC_OK);
+
+        FolderResponse createFolderResponse = response.as(FolderResponse.class);
+
+        createdFolderResponse = createFolderResponse;
+
+        assertNotNull(createFolderResponse.getId());
+        assertEquals(createFolderResponse.getName(), createFolderRequest.getName());
+        assertEquals(createFolderResponse.getParentFolderId(), createFolderRequest.getParentFolderId());
+        assertEquals(createFolderResponse.getBucketId(), createFolderRequest.getBucketId());
+    }
+
+    @Test(dependsOnMethods = "logged_owner_should_be_able_to_create_folder_in_its_own_private_bucket")
+    void logged_owner_should_be_able_to_upload_file_in_its_own_private_bucket() {
+        Response response = skyNestBackendClient.uploadFile(createdBucketResponse.getBucketId());
+        response.then().statusCode(SC_OK);
+
+        FileResponse uploadFileResponse = response.as(FileResponse.class);
+
+        uploadedFileResponse = uploadFileResponse;
+
+        assertEquals(uploadFileResponse.getName(), "FileForUpload.txt");
```

Comment:
although this probably works; it would be preferable to create the file dynamically, with a unique name, for better tracking purposes